core/chain-configs/src/genesis_config.rs

```diff
@@ -264,6 +266,37 @@ impl GenesisRecords {
     }
 }
 
+pub struct GenesisJsonHasher {
+    pub digest: sha2::Sha256,
+}
+
+impl GenesisJsonHasher {
+    pub fn new() -> Self {
+        Self { digest: sha2::Sha256::new() }
+    }
+
+    pub fn process_config(&mut self, config: &GenesisConfig) {
+        let mut ser = Serializer::pretty(&mut self.digest);
+        config.serialize(&mut ser).expect("Error serializing the genesis config.");
+    }
+
+    pub fn process_record(&mut self, record: &StateRecord) {
+        let mut ser = Serializer::pretty(&mut self.digest);
+        record.serialize(&mut ser).expect("Error serializing the genesis record.");
+    }
+
+    pub fn process_genesis(&mut self, genesis: &Genesis) {
+        self.process_config(&genesis.config);
+        for record in genesis.records.as_ref() {
+            self.process_record(record)
+        }
+    }
+
+    pub fn get_hash(self) -> CryptoHash {
+        CryptoHash(self.digest.finalize().into())
```

Comment:
what's the complexity of this function? Is `finalize` an expensive operation?

Reply:
nvm didn't realize that it takes `self` instead of `&self`

Reply:
Complexity stays the same as before because this function can only be called once. Although it's not clear from the name, I'll change that.